druid/src/widget/textbox.rs

```diff
@@ -90,8 +99,17 @@ impl TextBox {
             .build()
             .unwrap();
 
+        let text = if self.password_mode {
+            let mut t = String::new();
+            for _ in text.grapheme_indices(true) {
+                t.push('*');
+            }
+            t
```

Comment:
not a blocker but I would write this as,

```rust
let text = if self.password_mode {
    text.grapheme_indices(true).map(|_| '*').collect()
} else {
    text.to_string()
};
```

or even (because I think it's a bit annoying that we're allocating a `String` from a `&str` and then immediately borrowing it again):

```rust
static PASSWORD_CHARS: &str = "*****************************************";

let text = if self.password_mode {
    let num_chars = text.grapheme_indices(true).count().min(PASSWORD_CHARS.len());
    &PASSWORD_CHARS[..num_chars]
} else {
    text
};
```